analytics/management/commands/populate_analytics_db.py

```diff
@@ -94,17 +101,18 @@ def handle(self, *args: Any, **options: Any) -> None:
             role=UserProfile.ROLE_REALM_OWNER,
             force_date_joined=installation_time,
         )
-        do_change_user_role(shylock, UserProfile.ROLE_REALM_OWNER, acting_user=None)
+        UserGroupMembership.objects.create(user_profile=shylock, user_group=owners_system_group)
 
         # Create guest user for set_guest_users_statistic.
-        create_user(
+        bassanio = create_user(
             "bassanio@analytics.ds",
             "Bassanio",
             realm,
             full_name="Bassanio",
             role=UserProfile.ROLE_GUEST,
             force_date_joined=installation_time,
         )
+        UserGroupMembership.objects.create(user_profile=bassanio, user_group=guests_system_group)
```

Comment:
Can we have this code instead call a different function to create all the system groups, and use `do_create_user` to create the users without duplicating code here? I feel like we should be moving towards creating real objects rather than copying user/group creation logic into this file.

Reply:
We can try (if it doesn't end up breaking some more subtle initial setup assumption for the analytics realm), but `do_create_users` doesn't expose `force_date_joined` kwargs unlike `create_users`. We can add that of course, at the small cost of cluttering its api a bit more.

Does that sounds worth to proceed?

Reply:
It probably would be but it's pretty low importance and we've got a lot going on, so we can just do it when we're next messing with this code path not in a PR we're hoping to backport soon.